src/main/java/net/dv8tion/jda/internal/entities/WebhookImpl.java

```diff
@@ -68,20 +77,22 @@ public WebhookType getType()
     @Override
     public JDA getJDA()
     {
-        return channel.getJDA();
+        return api;
     }
 
     @Nonnull
     @Override
     public Guild getGuild()
     {
-        return channel.getGuild();
+        return getChannel().getGuild();
```

Comment:
Since `getChannel()` can now throw `IllegalStateException`, this should probably manually check `channel == null` and have a Guild-specific `IllegalStateException` instead of the channel-specific one bubbling up instead.